### PR TITLE
feat(skeleton): add skeleton loading component

### LINKED ISSUE #3105: Skeleton loading
Introduce support for Skeleton loading. Design document here: https://marvelapp.com/61e541a/screen/69171533

cc: @LHinson @mcoker

### PR DESCRIPTION
**Pull request by christiemolloy**
closes #3105

### CODE REVIEW

File: src/patternfly/components/Skeleton/examples/Skeleton.md [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+  {{#> skeleton-element skeleton-element--modifier="pf-m-rectangle-lg"}}
+  {{/skeleton-element}}
+{{/skeleton}}
+```
+
+```hbs title=Skeleton-container
+
+<div>4px spacing</div>
+{{#> skeleton}}
+  {{#> skeleton-container skeleton-container--modifier="pf-m-spacer-sm"}}
```
Comment (mcoker) [inline src/patternfly/components/Skeleton/examples/Skeleton.md:137]: did you mean to use xs here?

```suggestion
  {{#> skeleton-container skeleton-container--modifier="pf-m-spacer-xs"}}
```

File: src/patternfly/components/Skeleton/skeleton.scss [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+  // Container
+  --pf-c-skeleton__container--GridGap: 0;
+  --pf-c-skeleton__container--PaddingTop: 0;
+  --pf-c-skeleton__container--PaddingBottom: 0;
+
+  // Element
+  --pf-c-skeleton__element--BackgroundColor: var(--pf-global--palette--black-150);
+  --pf-c-skeleton__element--Height: var(--pf-global--spacer--lg);
+  --pf-c-skeleton__element--Width: 100%;
+  --pf-c-skeleton__element--BorderRadius: #{pf-size-prem(4px)}; // add variable ?
```
Comment (mcoker) [inline src/patternfly/components/Skeleton/skeleton.scss:16]: Seems like we should be able to use one of our existing border-radius vars?

Comment (christiemolloy) [reply src/patternfly/components/Skeleton/skeleton.scss:16]: @doruskova would you be ok with using a 3px border radius to align with the system?

Comment (doruskova) [reply src/patternfly/components/Skeleton/skeleton.scss:16]: @christiemolloy sure, no problem. I tried it in my codepen and the difference is minimal. Keep consistency with the system :)

File: src/patternfly/components/Skeleton/skeleton.scss [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+  --pf-c-skeleton__element--m-width-35--Width: 35%;
+  --pf-c-skeleton__element--m-width-40--Width: 40%;
+  --pf-c-skeleton__element--m-width-45--Width: 45%;
+  --pf-c-skeleton__element--m-width-50--Width: 50%;
+  --pf-c-skeleton__element--m-width-60--Width: 60%;
+  --pf-c-skeleton__element--m-width-70--Width: 70%;
+  --pf-c-skeleton__element--m-width-80--Width: 80%;
+  --pf-c-skeleton__element--m-width-90--Width: 90%;
+
+  &.pf-m-padding {
```
Comment (mcoker) [inline src/patternfly/components/Skeleton/skeleton.scss:58]: do we need the parent element, and this modifier?

File: src/patternfly/components/Skeleton/skeleton.scss [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+  }
+
+  &.pf-m-width-45 {
+    --pf-c-skeleton__element--Width: var(--pf-c-skeleton__element--m-width-45--Width);
+  }
+
+  &.pf-m-width-50 {
+    --pf-c-skeleton__element--Width: var(--pf-c-skeleton__element--m-width-50--Width);
+  }
+
```
Comment (mcoker) [inline src/patternfly/components/Skeleton/skeleton.scss:159]: It's kind of odd we don't do +5% increments after 50%. We could also use the same widths that are in the sizing utility - https://github.com/patternfly/patternfly/blob/61898c01f7dcbb7937d9ba624ab374166f9d96cb/src/patternfly/utilities/Sizing/sizing.scss#L5-L14

File: src/patternfly/components/Skeleton/skeleton.scss [hunk 5]
```diff
@@ -0,0 +1,217 @@
+.pf-c-skeleton {
+  // Padding modifier
+  --pf-c-skeleton--m-padding--Padding: var(--pf-global--spacer--lg);
+
+  // Element
+  --pf-c-skeleton__element--BackgroundColor: var(--pf-global--palette--black-150);
+  --pf-c-skeleton__element--Height: var(--pf-global--spacer--lg);
+  --pf-c-skeleton__element--Width: 100%;
```
Comment (mcoker) [inline src/patternfly/components/Skeleton/skeleton.scss:8]: do we need this?

File: src/patternfly/components/Skeleton/skeleton.scss [hunk 6]
```diff
@@ -0,0 +1,217 @@
+.pf-c-skeleton {
+  // Padding modifier
+  --pf-c-skeleton--m-padding--Padding: var(--pf-global--spacer--lg);
+
+  // Element
+  --pf-c-skeleton__element--BackgroundColor: var(--pf-global--palette--black-150);
+  --pf-c-skeleton__element--Height: var(--pf-global--spacer--lg);
```
Comment (mcoker) [inline src/patternfly/components/Skeleton/skeleton.scss:7]: Should this just be the current font size * line height maybe? I think that would make a good default.

File: src/patternfly/components/Skeleton/skeleton.scss (hunk 3 above)
Comment (christiemolloy) [reply src/patternfly/components/Skeleton/skeleton.scss:58]: True, I was thinking for elements that didnt have padding. But maybe we remove this and if there is a use case where the component needs padding we could add it back in

File: src/patternfly/components/Skeleton/skeleton.scss (hunk 4 above)
Comment (christiemolloy) [reply src/patternfly/components/Skeleton/skeleton.scss:159]: Yeah I was following what was in the table. But Ill follow the sizing utility instead and maybe one day we can update the table

File: src/patternfly/components/Skeleton/skeleton.scss (hunk 6 above)
Comment (christiemolloy) [reply src/patternfly/components/Skeleton/skeleton.scss:7]: Yep and that aligns with the text modifiers - sounds good

**Pull request closed without merging**